lib/IRGen/IRGenSIL.cpp

```diff
@@ -3629,8 +3629,6 @@ void IRGenSILFunction::visitRetainValueInst(swift::RetainValueInst *i) {
 }
 
 void IRGenSILFunction::visitRetainValueAddrInst(swift::RetainValueAddrInst *i) {
-  assert(i->getAtomicity() == RefCountingInst::Atomicity::Atomic &&
-         "Non atomic retains are not supported");
   SILValue operandValue = i->getOperand();
   Address addr = getLoweredAddress(operandValue);
   SILType addrTy = operandValue->getType();
```

Comment:
You should also thread through the atomicity parameter to `getOrCreateRetainFunction`. (Similar for the release equivalent).

```
llvm::Constant *IRGenModule::getOrCreateRetainFunction(const TypeInfo &ti,      
                                                       SILType t,               
                                                       llvm::Type *llvmType,
                                                       Atomicity atomicity) {
...
        loadableTI->copy(IGF, loaded, out, atomicity); 
```